### DISCUSSION TITLE
Want to be able to client-side cache `.pageContext.json`.

### DISCUSSION THREAD

**Discussion by hato810424** [vikejs/vike#1698] [Discussion]
### Description

Wouldn't the fetch be performed "every time" on authentication pages and on pages where `+data.ts` is used, leading to server and database load depending on the functionality? We wondered.

So I would like a mechanism to cache files such as `.pageContext.json`!

**Comment by brillout** [comment]
Yes, by default, it‘s fetched every time upon page navigation.

One thing you can try is to make the hook isomorphic so that Vike stops doing pageContext.json requests upon navigation. You‘ll then be able to use a cache on the client-side.

One thing Vike could do to ease that use case is to support two variants of the same hook:
~~~
pages/some-page/+data.server.js
pages/some-page/+data.client.js
~~~

**Reply by hato810424** [reply]
What happens to `+guard.js` in that case?

**Reply by brillout** [reply]
I don't know, I guess it would be forbidden.
